test/units/modules/network/nxos/test_nxos_interface.py

```diff
@@ -89,3 +93,49 @@ def test_nxos_interface_speed_idempotence(self):
         set_module_args(dict(interface='Ethernet2/1', speed='1000'))
         result = self.execute_module(changed=False)
         self.assertEqual(result['commands'], [])
+
+    def test_nxos_interface_bfd_1(self):
+        # enable / disable part 1
+        set_module_args(dict(
+            interface='Ethernet2/1',
+            mode='layer3',
```

Comment:
Why does the mode matter for this test?

Reply:
Idempotence, mostly. It's an issue with switching from L2 to L3 and vice versa, where the default admin state may change. 

If you don't specify `mode` in the playbook, then `mode` defaults to `None`, which causes another piece of code to think there's a mode change since the `want/have` values don't match; that in turn triggers add'l config to do a `<no> shutdown` (look for `commands2` in the module. To avoid the extra `shutdown` config I just specified `mode` in the playbook.

But now that you bring this up I see an easy fix so I'll just make it smarter about detecting `mode` changes.